src/Symfony/Component/HttpFoundation/File/MimeType/FileBinaryMimeTypeGuesser.php

```diff
@@ -43,7 +43,33 @@ public function __construct($cmd = 'file -b --mime %s 2>/dev/null')
      */
     public static function isSupported()
     {
-        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg');
+        static $supported;
+
+        if (isset($supported)) {
+            return $supported;
+        }
+
+        if ('\\' === DIRECTORY_SEPARATOR || !function_exists('passthru') || !function_exists('escapeshellarg')) {
+            return $supported = false;
+        }
+
+        ob_start();
+
+        passthru('command -v file', $return);
+        if ($return > 0) {
+            ob_end_clean();
+
+            return $supported = false;
+        }
+
+        $binPath = trim(ob_get_clean());
+
+        if (!preg_match('#^(?:/[^\0/])+#', $binPath)) {
```

Comment:
why this specific regex? can the NUL char actually happen?
also, it looks like this could be just `'#^/[^\0/]#'`

Reply:
Yes, it's possible to have a NUL character:

```php
ob_start();
passthru('find -maxdepth 1 -print0');
$output = trim(ob_get_clean());
echo ord($output[1]);
```

gives:

```
0
```